Add missing type hints in `core/dbt/task/clean.py`

### Housekeeping

- [X] I am a maintainer of dbt-core

### Short description

We're putting some effort into improving our typing. There are two type errors in `core/dbt/task/clean.py`, we should fix them.

```
core/dbt/task/clean.py:19: error: Function is missing a return type annotation  [no-untyped-def]
core/dbt/task/clean.py:19: note: Use "-> None" if function does not return a value
core/dbt/task/clean.py:49: error: Call to untyped function "warn" in typed context  [no-untyped-call]
```

### Acceptance criteria

Fix the typing issues

### Suggested Tests

N/A

### Impact to Other Teams

None

### Will backports be required?

No

### Context

_No response_

--- a/.changes/unreleased/Under the Hood-20240827-114810.yaml
+++ b/.changes/unreleased/Under the Hood-20240827-114810.yaml
@@ -1,0 +1,6 @@
+kind: Under the Hood
+body: Fix type errors in `dbt/core/task/clean.py`
+time: 2024-08-27T11:48:10.438173-05:00
+custom:
+  Author: QMalcolm
+  Issue: "10616"

--- a/core/dbt/deprecations.py
+++ b/core/dbt/deprecations.py
@@ -138,7 +138,7 @@ def renamed_env_var(old_name: str, new_name: str):
     return cb
 
 
-def warn(name, *args, **kwargs):
+def warn(name: str, *args, **kwargs) -> None:
     if name not in deprecations:
         # this should (hopefully) never happen
         raise RuntimeError("Error showing deprecation warning: {}".format(name))

--- a/core/dbt/task/clean.py
+++ b/core/dbt/task/clean.py
@@ -16,7 +16,7 @@ class CleanTask(BaseTask):
         self.config = config
         self.project = config
 
-    def run(self):
+    def run(self) -> None:
         """
         This function takes all the paths in the target file
         and cleans the project paths that are not protected.

